test/test_typed_array.cpp

```diff
@@ -0,0 +1,309 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <compare>
+#include <cstddef>
+#include <cstdint>
+#include <iostream>  // For Doctest
+#include <sstream>
+#include <string>
+#include <type_traits>
+
+#include "sparrow/typed_array.hpp"
+
+#include "array_data_creation.hpp"
+#include "doctest/doctest.h"
+
+using namespace sparrow;
+
+namespace
+{
+    template <typename O, std::integral I>
+    constexpr O to_value_type(I i)
+    {
+        if constexpr (std::is_same_v<O, numeric::float16_t>)
+        {
+            return static_cast<float>(i);
+        }
+        else if constexpr (std::is_arithmetic_v<O>)
+        {
+            return static_cast<O>(i);
+        }
+        else if constexpr (std::is_same_v<O, std::string>)
+        {
+            return std::to_string(i);
+        }
+    }
+
+    constexpr size_t n = 10;
+    constexpr size_t offset = 1;
+    const std::vector<size_t> false_bitmap = {9};
+}
+
+TEST_SUITE("typed_array")
+{
+    TEST_CASE_TEMPLATE_DEFINE("all", T, all)
+    {
+        SUBCASE("constructor with parameter")
+        {
+            constexpr size_t n = 10;
+            constexpr size_t offset = 1;
+            const auto array_data = sparrow::test::make_test_array_data<T>(n, offset);
+            const typed_array<T> typed_array(array_data);
```

Comment:
Consider using `ta{array_data}` instead of parenthesis so that we benefit from no-implicit-conversion-allowed (same for all functions below). I suspect we will not provide initializer list at the typed_array level, but if you think we will then ignore my remark.